<launch>
  <arg name="map_name" default="lab121_no_lidar"/>

  <!-- Run the point cloud and waypoint publishers for the map -->
  <include file="$(find rbd_spot_perception)/launch/graphnav_map_publisher_with_waypoints.launch">
    <arg name="map_name" value="$(arg map_name)"/>
    <arg name="output" value="log" />
  </include>

  <!-- Stream the robot's GraphNav pose as tf transforms and messages -->
  <node pkg="rbd_spot_perception" type="stream_graphnav_pose.py"
        name="stream_graphnav_pose"
        args="--pub-tf --pub-pose" output="log"/>

  <!-- Run the grid map projection script -->
  <node pkg="sloop_ros" type="spot_graphnav_cloud_to_grid_map.py"
        name="graphnav_pcd_to_grid_map"
        args="--name $(arg map_name)" output="screen"/>

  <!-- run the POMDP ROS interface for SLOOP MOS -->
  <node name="run_sloop_mos_spot"
        pkg="sloop_ros" type="run_agent" output="screen">
    <!-- So strange. bug you can't do ~config -->
    <rosparam param="config" command="load" file="$(find sloop_ros)/config/sloop_mos_spot_cit_first_floor.yaml"/>
    <param name="~pomdp_name" value="sloop_mos_spot"/>
    <remap from="~observation/grid_map"  to="/graphnav_gridmap"/>
    <remap from="~observation/robot_pose" to="/spot_body_pose"/>
  </node>

</launch>
